src/App.js

```diff
@@ -1,8 +1,58 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import "./App.css";
 
+import Button from './components/Button'
+import UserList from './components/UserList'
+import UserCard from "./components/UserCard";
+
 function App() {
-  return <div>Good luck dear students!</div>;
+  const [usersNames, setUsersNames] = useState([])
+  const [users, setUsers] = useState({})
+  const [currentUser, setCurrentUser] = useState({})
+  const [isLoading, setIsLoading] = useState(true)
+
+  useEffect(() => {
+    handleUsers()
+  }, [])
+
+  const handleUsers = async() => {
+    setUsersNames([])
+    setUsers({})
+    setCurrentUser({})
+    setIsLoading(true)
+
+    await fetch('https://randomuser.me/api/?results=5')
+      .then(res => res.json())
+        .then(data => {
+          setUsers(users => data)
+          setCurrentUser(currentUser => data.results[0])
+          for (let i in data.results) {
+            const user = {
+              name: `${data.results[i].name.first} ${data.results[i].name.last}`,
+            }
+
+            setUsersNames(usersNames => [...usersNames, user])
+          }
```

Comment:
Nice that you do this calculation of the name here but I would keep 2 things in the state that have similar structure (Users and usersNames). I would enhance the Users array before storing to the state by changing the name property for each user or I would create an entirely different user with only the properties that I want. 
Also the proper function to use when you want altered data from an array with the same number of items is the map function.